gateway/ln-gateway/src/lib.rs

```diff
@@ -523,66 +584,100 @@ impl Gateway {
             contract_id,
         } = payload;
 
-        let actor = self.select_actor(federation_id).await?;
-        let (outpoint, preimage) = actor.pay_invoice(contract_id).await?;
-        actor
-            .await_outgoing_contract_claimed(contract_id, outpoint)
-            .await?;
-        Ok(preimage)
+        let client = self.select_client(federation_id).await?;
+        let operation_id = client.gateway_pay_bolt11_invoice(contract_id).await?;
+        let mut updates = client
+            .gateway_subscribe_ln_pay(operation_id)
+            .await?
+            .into_stream();
+
+        while let Some(update) = updates.next().await {
+            match update {
+                GatewayExtPayStates::Success { preimage } => return Ok(preimage),
+                GatewayExtPayStates::Fail => {
+                    return Err(GatewayError::Other(anyhow!("Payment failed")))
+                }
+                GatewayExtPayStates::Canceled => {
+                    return Err(GatewayError::Other(anyhow!("Outgoing contract canceled")))
+                }
+                _ => {}
+            };
+        }
+
+        return Err(GatewayError::Other(anyhow!(
+            "Unexpected error occurred while paying the invoice"
+        )));
     }
 
     async fn handle_balance_msg(&self, payload: BalancePayload) -> Result<Amount> {
-        self.select_actor(payload.federation_id)
+        Ok(self
+            .select_client(payload.federation_id)
             .await?
             .get_balance()
-            .await
+            .await)
     }
 
     async fn handle_address_msg(&self, payload: DepositAddressPayload) -> Result<Address> {
-        self.select_actor(payload.federation_id)
+        let (_, address) = self
+            .select_client(payload.federation_id)
             .await?
-            .get_deposit_address()
-            .await
-    }
-
-    async fn handle_deposit_msg(&self, payload: DepositPayload) -> Result<TransactionId> {
-        let DepositPayload {
-            txout_proof,
-            transaction,
-            federation_id,
-        } = payload;
-
-        self.select_actor(federation_id)
-            .await?
-            .deposit(txout_proof, transaction)
-            .await
+            .get_deposit_address(now() + Duration::from_secs(86400 * 365))
+            .await?;
+        Ok(address)
     }
 
-    async fn handle_withdraw_msg(&self, payload: WithdrawPayload) -> Result<TransactionId> {
+    async fn handle_withdraw_msg(&self, payload: WithdrawPayload) -> Result<Txid> {
```

Comment:
Should this return the fedimint `TransactionId` or the bitcoin `Txid`?

Reply:
Bitcoin transaction ID is far more useful. When withdrawing I'd pop this into mempool.space or something to watch for confirmations.

Maybe it would be good to return both? But let's refine that once we get rid of that `run` loop and all the oneshot sender stuff.